src/Orchard.Cms.Web/Modules/Orchard.Users/Controllers/AccountController.cs

```diff
@@ -125,7 +134,7 @@ private IActionResult RedirectToLocal(string returnUrl)
             }
             else
             {
-                return Redirect("~/");
+                return Redirect("~/" + _shellSettings.RequestUrlPrefix);
```

Comment:
I don't want to add unnecessary helper methods, even more that we'd still need to pass in the shellsettings. I still have hope we can make it automatic too using IHostingEnvironment.BasePath if we set it correctly and register a custom one in the shell service provider.